.github/workflows/dev.yml

```diff
@@ -55,9 +60,18 @@ jobs:
         run: pytest --ddtrace test/integration-test --module-path dist/lib/mod_datadog.so --bin-path /httpd/httpd-build/bin/apachectl --log-dir $(pwd)/logs -m smoke
       - name: Run Integration Tests
         run: pytest --ddtrace test/integration-test --module-path dist/lib/mod_datadog.so --bin-path /httpd/httpd-build/bin/apachectl --log-dir $(pwd)/logs -m ci
-      # - name: Upload coverage reports to Codecov
-      #   uses: codecov/codecov-action@v4.0.1
-      #   with:
-      #     token: ${{ secrets.CODECOV_TOKEN }}
-      #     slug: DataDog/httpd-datadog
+      - name: Import GNCO files
+        uses: actions/download-artifact@v3
+        with: 
+          name: gcno-artifacts
+          path: build/mod_datadog/CMakeFiles/mod_datadog.dir/src
```

Comment:
`.gcno` are metadata on the source code generated at compilation. It's mainly there to helps determining where coverage data should be collected.

On the other hand, `.gcda` holds the actual coverage data from the program's execution.

Reply:
Is the purpose of this comment to say it lacks the .gcda ? Because from what i understood to compute the coverage we need both the .gnco and the .gcda

The .gnco files are generated at compilation as you said so we lose them in the next jobs, that is why i upload them. On the other side, the .gcda we'll be automatically generated in the same folder (as the .gnco files) after the tests. Therefore, we do have the gcda files when I call the lcov command.

Reply:
Indeed. And you had to do all the shenanigans because the module is not built in this step 🤦‍♂️
My apologies than.